src/library_memfs.js

```diff
@@ -192,18 +192,18 @@ addToLibrary({
         return MEMFS.createNode(parent, name, mode, dev);
       },
       rename(old_node, new_dir, new_name) {
-        // if we're overwriting a directory at new_name, make sure it's empty.
-        if (FS.isDir(old_node.mode)) {
-          var new_node;
-          try {
-            new_node = FS.lookupNode(new_dir, new_name);
-          } catch (e) {
-          }
-          if (new_node) {
+        var new_node;
+        try {
+          new_node = FS.lookupNode(new_dir, new_name);
+        } catch (e) {}
+        if (new_node) {
+          if (FS.isDir(old_node.mode)) {
+            // if we're overwriting a directory at new_name, make sure it's empty.
             for (var i in new_node.contents) {
               throw new FS.ErrnoError({{{ cDefs.ENOTEMPTY }}});
             }
           }
+          FS.hashRemoveNode(new_node);
```

Comment:
Actually can we move this whole `if` block inside the `try` block and then remove the `if`?

Reply:
Sure. Though these blanket catches make debugging kind of hard. It would be nice to have a variant of `lookupNode` that returns `undefined` if the node is missing...

Reply:
Wait, we can't throw inside the catch block so that doesn't work.

Reply:
Why can't we throw inside that catch block?

Reply:
> Sure. Though these blanket catches make debugging kind of hard. It would be nice to have a variant of `lookupNode` that returns `undefined` if the node is missing...

Maybe we should do that first?

Reply:
Maybe we can clean this up as a followup?